server/src/vote.rs

```diff
@@ -79,9 +87,17 @@ pub(super) async fn vote(
                 .attributes()
                 .and_then(|a| a.get("votes"))
                 .and_then(|v| v.as_n().ok())
-                .and_then(|v| v.parse::<isize>().ok());
+                .and_then(|v| v.parse::<usize>().ok());
             Ok(Json(serde_json::json!({ "votes": new_count })))
         }
+        Err(ref error @ SdkError::ServiceError(ref e)) => {
+            if e.err().is_conditional_check_failed_exception() {
```

Comment:
Would be nice if we could check specifically that it's the zero condition that failed, just for the sake of future compatibility in case we add others, but this is fine for now :+1: